pkg/util/kubernetes/apiserver/events.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+//// Covered by test/integration/util/kube_apiserver/events_test.go
+
+import (
+	"context"
+	"io"
+	"strconv"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/ericchiang/k8s"
+	"github.com/ericchiang/k8s/api/v1"
+)
+
+var eventReadTimeout = 100 * time.Millisecond
+
+// LatestEvents retrieves all the cluster events happening after a given token.
+// First slice is the new events, second slice the modified events.
+// If the `since` parameter is empty, we query the apiserver's cache to avoid
+// overloading it.
+// https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#watch-list-289
+func (c *APIClient) LatestEvents(since string) ([]*v1.Event, []*v1.Event, string, error) {
+	var addedEvents, modifiedEvents []*v1.Event
+	latestResVersion := &since
+
+	var sinceOption k8s.Option
+	if len(since) > 0 {
+		sinceOption = k8s.ResourceVersion(since)
+	} else {
+		// Only retrieve what is in the apiserver cache.
+		// Else we'll get one hour worth of events.
+		sinceOption = k8s.ResourceVersion("")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+
+	watcher, err := c.client.CoreV1().WatchEvents(ctx, "", sinceOption)
+	if err != nil {
+		return addedEvents, modifiedEvents, since, err
+	}
+
+	// Start a timer that will cancel the connection when the watcher is idle.
+	// That allows `watcher.Next()` to return context.Canceled and break the reading loop.
+	timeout := time.AfterFunc(eventReadTimeout, func() { cancel() })
+
+	for {
+		meta, event, err := watcher.Next()
+		if err != nil {
+			if err != context.Canceled && err != io.EOF {
+				log.Debugf("stopping event collection, got error: %s", err)
+			} // else silently stop
+			break
+		}
+		timeout.Reset(eventReadTimeout)
+
+		if event != nil && event.Metadata != nil && event.Metadata.ResourceVersion != nil {
+			resVersionMetadata, err := strconv.Atoi(*event.Metadata.ResourceVersion)
+			resVersionCached, _ := strconv.Atoi(*latestResVersion)
```

Comment:
`latestResVersion` come from string *since*, why ignoring the error ?

Reply:
So I figured this was something we initialised as a string and converted to a string before. I assumed an error would be redundant with:
https://github.com/DataDog/datadog-agent/blob/ac11dc23f8d0c70fe93beef305f60f866f19f3bb/pkg/collector/corechecks/cluster/kubernetes_apiserver.go#L97

Reply:
that logic might change sometime, might as well check indeed